gnucash/import-export/aqb/gnc-flicker-gui.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * gnc-flicker-gui.c --
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, contact:
+ *
+ * Free Software Foundation           Voice:  +1-617-542-5942
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org
+ */
+
+/**
+ * @internal
+ * @file gnc-flicker-gui.c
+ * @brief GUI callbacks for Flicker and ChipTAN(optisch)
+ * @author Copyright (C) 2020 Christian Wehling <christian.wehling@web.de>
+ */
+
+#include <cairo.h>
+#include <gtk/gtk.h>
+#include <glib/gi18n.h>
+
+#include "gnc-flicker-gui.h"
+
+/* structured data for the flicker variables */
+struct _flickerdraw {
+	const char *challenge;
+	guint maxWert;
+    guint margin;           /* Abstand zwischen Balken */
+    guint barwidth;         /* Balkenbreite */
+    guint barheight;        /* Balkenhöhe */
+    guint x_barpos;			/* x-Wert für die Position des Balken */
+    guint y_barpos;			/* y-Wert für die Position des Balken */
+    guint x_drawpos;		/* Erste Zeichenposition */
+    guint y_drawpos;		/* Erste Zeichenposition */
+    guint height;			/* Höhe der Zeichenfläche */
+    guint width;			/* Breite der Zeichenfläche */
+    guint delay;            /* Wartezeit zwischen Frames in Millisekunden */
+    guint halfbyteid;
+    guint clock;
+    guint interval;
+    gboolean change_interval;
+} flickerdraw;
+
+static GncFlickerGui *flickergui = NULL;
+_Bool bitarray[100][5];
```

Comment:
_Bool is not a type in GnuCash, and since the smallest possible addressable unit in C is an `unsigned char`/`uint8_t`, this takes at least 500 bytes to store 500 bits. Use `uint8_t bitarray[100]` and bitwise operators instead.